ledger/complete/wal/compactor.go

```diff
@@ -24,48 +27,51 @@ func NewCompactor(checkpointer *Checkpointer, interval time.Duration, checkpoint
 	}
 	return &Compactor{
 		checkpointer:       checkpointer,
-		done:               make(chan struct{}),
 		stopc:              make(chan struct{}),
+		observers:          make(map[observable.Observer]struct{}),
+		lm:                 lifecycle.NewLifecycleManager(),
 		interval:           interval,
 		checkpointDistance: checkpointDistance,
 		checkpointsToKeep:  checkpointsToKeep,
 	}
 }
 
-// Ready periodically fires Run function, every `interval`
-// If called more then once, behaviour is undefined.
-func (c *Compactor) Ready() <-chan struct{} {
-	ch := make(chan struct{})
+func (c *Compactor) Subscribe(observer observable.Observer) {
+	if _, exists := c.observers[observer]; !exists {
+		var void struct{}
+		c.observers[observer] = void
```

Comment:
Would this behave exactly the same if we just skip the the if statement and run this single line? Seems regardless of whether `exists` is true or false we need to perform the lookup.